// Copyright (c) Alex Jamieson-Binnie. All rights reserved.
// Licensed under the GPL. See License.txt in the project root for license information.

syntax = "proto3";

import "google/protobuf/struct.proto";

package narupa.protocol.renderer;

message CreateCameraRequest {
  string id = 1;
  google.protobuf.Struct properties = 2;
}

message CreateCameraResponse {
  string id = 1;
}

message UpdateCameraRequest {
  string id = 1;
  google.protobuf.Struct properties = 2;
}

message UpdateCameraResponse {
  string id = 1;
}

service CameraService {
  rpc CreateCamera (CreateCameraRequest) returns (CreateCameraResponse);
  rpc UpdateCamera (UpdateCameraRequest) returns (UpdateCameraResponse);
}
